library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+      # @param headline [String, :error, :warning] sets popup headline. When String is passed it is shown as it is.
+      #   When empty string is passed no headline is shown.
+      #   If Symbol is passed it have to be from predefined set of symbols.
+      #   Note: Symbol it is just predefined strings, does not modify style of popup.
+      # @param timeout [Integer] how long wait till autoclose dialog. 0 means wait forever.
+      # @param buttons [Hash<Symbol, String>, Symbol] specified which buttons popup will have.
+      #   it can be hash in format button_id => button_text, showed in same order as in hash.
+      #   Beware that :details and :stop id is reserved.
+      #   The second option is symbol that specify one of predefined set of buttons.
+      #   Current set of predefined buttons are
+      #   `:ok` -> `{ ok: Label.OKButton}`
+      #   `continue_cancel` ->   `{ continue: Label.ContinueButton, cancel: Label.CancelButton }`
+      #   `:yes_no` -> `{ yes: Label.YesButton, no: Label.NoButton }`
+      # @param focus [Symbol, nil] what button focus.
+      #   Also it is button which is returned if timeout exceed.
+      #   if it is nil, then choose the first button. See buttons parameter.
+      # @param richtext [Boolean, :auto] if use richtext widget.
+      #   Useful when report contain richtext tags or is long, so it have to be scrollable.
+      #   If :auto is used, then it detect too long text and use richtext for it and! escape
+      #   richtext tags, so if text is really richtext then use true.
+      # @param style [:notice, :important, :warning] popup dialog styling. :notice is common one,
+      #    :important is brighter and :warning is style when something goes wrong.
+      #    See Yast::UI.OpenDialog options :infocolor and :warncolor.
+      # @return [Symbol] symbol of pressed button. If timeout appear,
+      #   then button set in focus parameter is returned. If user click on 'x' button in window
+      #   then `:cancel` symbol is returned.
+      #
+      # @example pair of old and new API calls
+      #   Yast::Popup.Message(text)
+      #   Yast2::Popup.show(text)
+      #
+      #   Yast::Popup.MessageDetails(text, details)
+      #   Yast2::Popup.show(text, details: details)
+      #
+      #   Yast::Popup.TimedError(text, seconds)
+      #   Yast2::Popup.show(text, headline: :error, timeout: seconds)
+      #
+      #   Yast::Popup.TimedErrorAnyQuestion(headline, message, yes_button_message, no_button_message, focus, timeout_seconds)
+      #   Yast2::Popup.show(message, headline: headline, timeout: timeout_seconds, buttons: { yes: yes_button_message, no: no_button_message), focus: :yes)
+
+      #   Yast::Popup.TimedLongNotify(message, timeout_seconds)
+      #   Yast2::Popup.show(message, richtext: true, timeout: timeout_seconds)
+      def show(message, details: "", headline: "", timeout: 0, focus: nil, buttons: :ok, richtext: :auto, style: :notice)
```

Comment:
Supporting `nil` for some arguments like `:details`, `:headline`, or `:timeout` (in addition to zero and empty strings) would not be much work and would be quite rubist (or maybe railsty). In any case, I'm sure it will not hurt.

Reply:
well, I explicitelly do not want to use nil. I would like to avoid it where possible as it is for me a bit error prone and require explicit type looks better for me and also `headline: ""` looks better for me then `headline: nil`